ignite/metrics/precision.py

```diff
@@ -41,32 +41,33 @@ def __init__(
     def reset(self) -> None:
         if self._is_multilabel:
             init_value = 0.0 if self._average else []
-            kws = {"dtype": torch.float64, "device": self._device}
-            self._true_positives = torch.tensor(init_value, **kws)
-            self._positives = torch.tensor(init_value, **kws)
+            self._true_positives = torch.tensor(init_value, dtype=torch.float64, device=self._device)
+            self._positives = torch.tensor(init_value, dtype=torch.float64, device=self._device)
         else:
             self._true_positives = 0
             self._positives = 0
 
         super(_BasePrecisionRecall, self).reset()
 
     def compute(self) -> Union[torch.Tensor, float]:
-        is_scalar = not isinstance(self._positives, torch.Tensor) or self._positives.ndim == 0
+        is_scalar = (
+            not isinstance(self._positives, torch.Tensor) or self._positives.ndim == 0  # type: ignore[attr-defined]
+        )
         if is_scalar and self._positives == 0:
             raise NotComputableError(
                 "{} must have at least one example before it can be computed.".format(self.__class__.__name__)
             )
 
         if not (self._type == "multilabel" and not self._average):
-            if not self._is_reduced:
-                self._true_positives = idist.all_reduce(self._true_positives)
-                self._positives = idist.all_reduce(self._positives)
+            if not self._is_reduced:  # type: ignore[has-type] # mypy cann't determine type
+                self._true_positives = idist.all_reduce(self._true_positives)  # type: ignore[arg-type, assignment]
```

Comment:
Maybe, mypy is right here that we should not reuse the same attribute for the output as we can just define a local vars initialized as `self._positives` or `self._true_positives` ?

Reply:
actually it is a problem related to the `numbers.Number` used in `all_reduce`
```python
def all_reduce(tensor: Union[torch.Tensor, Number], op: str = "SUM") -> Union[torch.Tensor, Number]:
```
there is still an open issue in `mypy` https://github.com/python/mypy/issues/3186 which suggests the usage of `float` because it would allow both `int` and `float`.
What do you think about replacing the `Number` type with `float` in the whole project? I could create a new issue to tackle that problem.

Reply:
@gruebel sounds good about replacing `Number` with `float` for type annotations (if I understood correctly). Runtime code can still keep Number, right? 
So, yes, please open an issue for that.

Reply:
Yeah correct it is just for type hinting 😄  and I created a new issue for it.

Reply:
Perfect, thanks!